client/blocks/comment-detail/style.scss

```diff
@@ -0,0 +1,261 @@
+%comment-detail__ellipsis {
+	overflow: hidden;
+	text-overflow: ellipsis;
+	white-space: nowrap;
+}
+
+%comment-detail__gray-link {
+	color: $gray;
+	cursor: pointer;
+	fill: $gray;
+	&:focus,
+	&:hover {
+		color: $blue-wordpress;
+		fill: $blue-wordpress;
+	}
+}
+
+.comment-detail.card {
+	font-size: 14px;
+	margin: 0 auto;
+	padding: 0;
+	transition: margin .15s linear;
+
+	&.is-expanded {
+		margin: 16px auto;
+	}
+	&:not( .is-expanded ):hover {
+		box-shadow: 0 0 0 1px $gray, 0 2px 4px lighten( $gray, 20% );
+		z-index: 1;
+	}
+
+	.gridicon {
+		margin-right: 4px;
+		vertical-align: middle;
+	}
+}
+
+.comment-detail__author-avatar {
+	flex-shrink: 0;
+	img {
+		border-radius: 50%;
+	}
+}
+.comment-detail__author-info {
+	@extend %comment-detail__ellipsis;
+	color: $gray;
+
+	strong {
+		color: $gray-text;
+		margin-right: 8px;
+	}
+}
+
+.comment-detail__header {
+	display: flex;
+	flex-flow: row nowrap;
+	justify-content: space-between;
+	padding: 12px 8px;
+
+	&.is-preview {
+		cursor: pointer;
+	}
+
+	.comment-detail.is-expanded & {
+		border-bottom: 1px solid lighten( $gray, 30% );
+	}
+
+	.comment-detail__author-avatar {
+		display: inline-block;
+		height: 24px;
+		padding: 0 8px;
+		vertical-align: middle;
+		width: 24px;
+	}
+
+	.comment-detail__comment-preview {
+		@extend %comment-detail__ellipsis;
+		padding: 0 8px;
+		width: 60%;
+	}
+
+	.comment-detail__actions {
+		color: $gray;
+		a {
+			@extend %comment-detail__gray-link;
+			padding: 8px;
+			white-space: nowrap;
+		}
+		span {
+			display: none;
+
+			@include breakpoint( ">960px" ) {
+				display: inline;
+			}
+		}
+	}
+}
+
+.comment-detail__post {
+	border-bottom: 1px solid lighten( $gray, 30% );
+	display: flex;
+	flex-flow: row nowrap;
+	justify-content: flex-start;
+	padding: 12px 16px;
+
+	.reader-avatar {
+		flex-shrink: 0;
+		margin-left: 0;
+	}
+
+	.comment-detail__post-info {
+		@extend %comment-detail__ellipsis;
+		color: $gray;
+		line-height: 32px;
+		padding-left: 4px;
+
+		span {
+			color: $gray-text;
+			margin-right: 8px;
+		}
+
+		a {
+			@extend %comment-detail__gray-link;
+		}
+	}
+}
+
+.comment-detail__comment {
+	border-bottom: 1px solid lighten( $gray, 30% );
+	padding-left: 16px;
+
+	.comment-detail__comment-content {
```

Comment:
Ah so as a general rule, we make our classnames so long so we can just target them directly. This leaves CSS specificity at (0, 1, 0) (ids, classes, elements), and makes it easy to override rules (which is often a problem in large application css files). Thee are exceptions of course, but mainly we only do so for component composition or modifier classes.

See also: https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md#csssass-coding-guidelines  
> Since every element will have a direct class to be selected with, we avoid descendant selectors and child selectors. We also avoid indenting selectors in Sass files in favor of single selectors. The main exceptions are precisely the is-modifier class, and the cases where the context needs to change a given component's accidents.